openssl-dynamic/src/main/c/tcn_lock.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2023 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef TCN_LOCK_H
+#define TCN_LOCK_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif // __cplusplus
+
+typedef void* tcn_lock_t;
+
+tcn_lock_t tcn_lock_new();
```

Comment:
nit: use `malloc`/`free` terminology or `new`/`delete`, since this is a `c` API maybe the former makes sense?